components/psst/browser/content/psst_tab_web_contents_observer.cc

```diff
@@ -181,7 +189,26 @@ void PsstTabWebContentsObserver::OnUserScriptResult(
   inject_script_callback_.Run(
       MaybeAddParamsToScript(std::move(rule),
                              std::move(user_script_result).TakeDict()),
-      base::DoNothing());
+      base::BindOnce(&PsstTabWebContentsObserver::OnPolicyScriptResult,
+                     weak_factory_.GetWeakPtr(), id));
+}
+
+void PsstTabWebContentsObserver::OnPolicyScriptResult(
+    int nav_entry_id,
+    base::Value script_result) {
+  const auto script_result_parsed =
+      psst_script_responses::PolicyScriptResult::FromValue(script_result);
+  if (!script_result_parsed || !ShouldInsertScriptForPage(nav_entry_id)) {
+    return;
+  }
+
+  ui_delegate_->SetAppliedItems(script_result_parsed->progress,
```

Comment:
I don't understand `progress` here. This is a once callback, the script can't keep calling it multiple times.

Reply:
but that's what I'm saying, you can't do that. It's a promise that gets resolved and it can only get resolved once. This is also a once callback which means you can only call it once. I don't see a test for a policy script that updates the progress by calling it more than once and it doesn't look like that test could work if it did exist.

Reply:
That test has two navigations in it so I'm confused about how this is supposed to work. This is a multi-step process that involves several user navigations?